components/core/cmake/Options/options.cmake

```diff
@@ -309,171 +191,89 @@ function(validate_clp_s_search_dependencies)
     )
 endfunction()
 
-function(set_clp_s_search_dependencies)
-    set_clp_need_flags(
-        CLP_NEED_ABSL
-        CLP_NEED_LOG_SURGEON
-        CLP_NEED_SIMDJSON
-        CLP_NEED_SPDLOG
-    )
-endfunction()
-
 function(validate_clp_s_search_ast_dependencies)
     validate_clp_dependencies_for_target(CLP_BUILD_CLP_S_SEARCH_AST
         CLP_BUILD_CLP_S_TIMESTAMPPATTERN
     )
 endfunction()
 
-function(set_clp_s_search_ast_dependencies)
-    set_clp_need_flags(
-        CLP_NEED_SIMDJSON
-    )
-endfunction()
-
 function(validate_clp_s_search_kql_dependencies)
     validate_clp_dependencies_for_target(CLP_BUILD_CLP_S_SEARCH_KQL
         CLP_BUILD_CLP_STRING_UTILS
         CLP_BUILD_CLP_S_SEARCH_AST
     )
 endfunction()
 
-function(set_clp_s_search_kql_dependencies)
-    set_clp_need_flags(
-        CLP_NEED_ANTLR
-        CLP_NEED_SPDLOG
-    )
-endfunction()
-
 function(validate_clp_s_search_sql_dependencies)
     validate_clp_dependencies_for_target(CLP_BUILD_CLP_S_SEARCH_SQL
         CLP_BUILD_CLP_S_SEARCH_AST
     )
 endfunction()
 
-function(set_clp_s_search_sql_dependencies)
-    set_clp_need_flags(
-        CLP_NEED_ANTLR
-        CLP_NEED_SPDLOG
-    )
-endfunction()
-
 function(validate_clp_s_timestamppattern_dependencies)
     validate_clp_dependencies_for_target(CLP_BUILD_CLP_S_TIMESTAMPPATTERN
         CLP_BUILD_CLP_STRING_UTILS
     )
 endfunction()
 
-function(set_clp_s_timestamppattern_dependencies)
-    set_clp_need_flags(
-        CLP_NEED_DATE
-        CLP_NEED_SPDLOG
-    )
-endfunction()
-
 # Validates that for each target whose `CLP_BUILD_` option is `ON`, the `CLP_BUILD_` options for
-# the target's dependencies are also `ON`; Sets the required `CLP_NEED_` flags for any target that
-# will be built.
-function(validate_and_setup_all_clp_dependency_flags)
+# the target's dependencies are also `ON`.
+function(validate_all_clp_dependency_flags)
     if (CLP_BUILD_EXECUTABLES)
```

Comment:
Update `validate_clp_dependencies_for_target` so that we can remove this `if (CLP_BUILD_EXECUTABLES)`. Ditto for all the following if blocks.